app.js

```diff
@@ -53,32 +55,40 @@ function truncateName(name, charsLimit){
   return name.slice(0, charsLimit) + '…';
 };
 
+function updateMenu(items){
+  for(var index in items) {
+    for(var key in items[index].template) {
+      menuTemplate[index][key] = items[index].template[key];
+    }
+
+    menuTemplate[index].visible = menuTemplate[index].label.length > 0;
+  }
+
+  contextMenu = Menu.buildFromTemplate(menuTemplate);
+
+  appTray.setContextMenu(contextMenu);
+};
+
 var currentName = '', currentState = 'paused';
+
 itunes.on('playing', function(data){
   currentState = 'playing';
   currentName = data.name;
   appTray.setTitle('▶ ' + truncateName(currentName) + '  ');
 
-  // TODO: Update Album and Artist menu items (Atom shell doesn't support this yet)
-  // var menuItems = contextMenu.items;
-  // var albumMenu = menuItems[0];
-  // var artistMenu = menuItems[1];
-  // if (data.album){
-  //   albumMenu.label = 'Album: ' + data.album;
-  //   albumMenu.visible = true;
-  // } else {
-  //   albumMenu.visible = false;
-  // }
-  // if (data.artist){
-  //   artistMenu.label = 'Artist: ' + data.artist;
-  //   artistMenu.visible = true;
-  // } else {
-  //   artistMenu.visible = false;
-  // }
+  updateMenu([
+    { index: 0, template: { label: (data.album)  ? 'Album: '  + data.album  : '' } },
+    { index: 1, template: { label: (data.artist) ? 'Artist: ' + data.artist : '' } }
+  ]);
 });
+
 itunes.on('paused', function(data){
   currentState = 'paused';
   appTray.setTitle(defaultTitle);
+  updateMenu([
+    { index: 0, template: { label: '' } },
+    { index: 1, template: { label: '' } }
```

Comment:
Perhaps can set `visible: false` instead of `label: ''`, which I think is more explicit?